test/quantization/quantize_/workflows/int4/test_int4_opaque_tensor.py

```diff
@@ -76,6 +78,43 @@ def test_module_path(self, dtype):
                 "<class 'torchao.quantization.Int4OpaqueTensor'>",
             )
 
+    def test_activation_prescaling(self):
+        dtype = torch.bfloat16
+        input = torch.randn(1, 128, dtype=dtype)
+        linear1 = torch.nn.Linear(128, 256, bias=False, dtype=dtype)
+        linear2 = torch.nn.Linear(128, 256, bias=False, dtype=dtype)
+        with torch.no_grad():
+            linear2.weight.copy_(linear1.weight)
+        original = linear2(input)
+        quantize_(linear1, get_config(group_size=128))
+        quantize_(linear2, get_config(group_size=128))
+        qw1 = linear1.weight
+        assert isinstance(qw1, SupportsActivationPreScaling), (
+            "Expected int4 tensor supports activation prescaling"
+        )
+        assert qw1.act_pre_scale is None, "Default `act_pre_scale` is None"
+
+        _ACT_PRE_SCALE = 2
+        manual_scaled_quantized = linear1(input * _ACT_PRE_SCALE)
+        qw2 = linear2.weight
+        qw2.act_pre_scale = _ACT_PRE_SCALE
+        auto_scaled_quantized = linear2(input)
+
+        # Making sure activation pre scaling is successfully applied to the activation.
+        # manual_scaled_quantized (input * 2 → quantize with act_pre_scale=None) should equal
+        # auto_scaled_quantized (original input → quantize with act_pre_scale=2),
+        # Proving that the act_pre_scale factor correctly applies input scaling
```

Comment:
```suggestion
        # Make sure activation pre scaling is successfully applied to the activation.
```
Let's make it more concise.

Reply:
Thanks, updated